File: spring-cli-app-java/src/main/java/com/steeplesoft/spring_cli_app_java/ExampleCommand.java
package com.steeplesoft.spring_cli_app_java;

import com.beust.jcommander.Parameter;
import com.beust.jcommander.Parameters;
import org.springframework.stereotype.Component;

@Component
@Parameters(commandNames = ExampleCommand.COMMAND_NAME,
        commandDescription = "Example command")
public class ExampleCommand implements Command {
    public static final String COMMAND_NAME = "example";

    @Parameter(names = "--example", description = "Example parameter")
    private String example;

    @Override
    public String commandName() {
        return COMMAND_NAME;
    }

    @Override
    public void run() {
        System.out.println("You ran the command " + COMMAND_NAME + " with the parameter --example set to " + example);
    }
}

File: spring-cli-app-kotlin/src/main/kotlin/com/steeplesoft/springcli/ExampleCommand.kt
package com.steeplesoft.springcli

import com.beust.jcommander.Parameter
import com.beust.jcommander.Parameters
import org.springframework.stereotype.Component

@Component
@Parameters(commandNames = [ExampleCommand.COMMAND_NAME], commandDescription = "Example command")
class ExampleCommand : Command {
    @Parameter(names = ["--example"], description = "Example parameter")
    private var example: String? = null

    override fun commandName(): String {
        return COMMAND_NAME
    }

    override fun run() {
        println("You ran the command $COMMAND_NAME with the parameter --example set to $example")
    }

    companion object {
        const val COMMAND_NAME = "example"
    }
}
